ext/opcache/ZendAccelerator.c

```diff
@@ -3019,28 +3019,40 @@ static void accel_move_code_to_huge_pages(void)
 		long unsigned int  start, end, offset, inode;
 		char perm[5], dev[10], name[MAXPATHLEN];
 		int ret;
-
-		while (1) {
-			ret = fscanf(f, "%lx-%lx %4s %lx %9s %lu %s\n", &start, &end, perm, &offset, dev, &inode, name);
-			if (ret == 7) {
+		extern char *__progname;
+		char buffer[MAXPATHLEN];
+
+		while (fgets(buffer, MAXPATHLEN, f)) {
+			ret = sscanf(buffer, "%lx-%lx %4s %lx %9s %lu %s\n", &start, &end, perm, &offset, dev, &inode, name);
```

Comment:
Thank you so much for your quick review!
I used /lib64/ld-linux-x86-64.so.2 to start php-fpm. It will pack all the binaries into 4GB memory. Therefore, the first line in `/proc/self/maps` is no longer php-fpm, like this:
```
7ffff2599000-7ffff263a000 rw-p 00000000 00:00 0
7ffff268b000-7ffff26bb000 rw-p 00000000 00:00 0
...
7ffff26bb000-7ffff2780000 r--p 00000000 00:3b 25823057                   /usr/lib/x86_64-linux-gnu/libcrypto.so.3
...
7ffff6e00000-7ffff71d0000 r-xp 00200000 00:3b 26061893                   /usr/local/sbin/php-fpm
...
```
Since the first line `name` is empty , if using `fscanf()`, it will treat all of the following lines as the the `name`.  gdb can explain this bug:
```
(gdb) p name
$1 = "7ffff268b000-7ffff26bb000\000\000\000\377\177\000\000\002\000\000\000\000\000\000\000\004\000\000\000\377\177\000\000\003\000\000\000\000\000\000\000\004\000\000\000\000\000\000\000\004\000\000\000\000\000\000\000\004\000\000\000\000\000\000\000\005\000\000\000\000\000\000\000\004\000\000\000\000\000\000\000\006\000\000\000\000\000\000\000\004\000\000\000\004\000\000\000\a\000\000\000\000\000\000\000\004\000\000\000\000\000\000\000\b\000\000\000\000\000\000\000\004\000\000\000\000\000\000\000\340\377?", '\000' <repeats 13 times>, "\377\a", '\000' <repeats 33 times>, "\001", '\000' <repeats 12 times>...
(gdb) p/x start
$2 = 0x7ffff2599000
(gdb) p/x end
$3 = 0x7ffff263a000
```
Using fgets+sscanf can avoid this corner case.

Reply:
Good catch!